docs/activities.md

```diff
@@ -263,17 +198,48 @@ If this timeout is reached, the Activity Task fails and a retry occurs if a [Ret
 An Activity Heartbeat is a ping from the Worker that is executing the Activity to the Temporal Cluster.
 Each ping informs the Temporal Cluster that the Activity Execution is making progress and the Worker has not crashed.
 
+- [How to Heartbeat an Activity](/application-development-guide#activity-heartbeats)
+
 Activity Heartbeats work in conjunction with a [Heartbeat Timeout](#heartbeat-timeout).
 
 Activity Heartbeats are implemented within the Activity Definition.
 Custom progress information can be included in the Heartbeat which can then be used by the Activity Execution should a retry occur.
 
 An Activity Heartbeat can be recorded as often as needed (e.g. once a minute or every loop iteration).
+It is often a good practice to Heartbeat on anything but the shortest Activity Function Execution.
 Temporal SDKs control the rate at which Heartbeats are sent to the Cluster.
 
 Heartbeating is not required from [Local Activities](#local-activities), and does nothing.
 
-- [How to Heartbeat an Activity in Go](/application-development-guide/#activity-heartbeats)
+For _long-running_ Activities, we recommend using a relatively short Heartbeat Timeout and a frequent Heartbeat.
+That way if a Worker fails it can be handled in a timely manner.
+
+A Heartbeat can include an application layer payload that can be used to _save_ Activity Execution progress.
+If an [Activity Task Execution](/tasks/#activity-task-execution) times out due to a missed Heartbeat, the next Activity Task can access and continue with that payload.
+
+### Asynchronous Activity Completion
+
+Asynchronous Activity Completion is a feature that enables an Activity Function to return without causing the Activity Execution to complete.
+The Temporal Client can then be used to both Heartbeat Activity Execution progress and eventually provide a result if needed.
+
+- [How to complete an Activity Asynchronously](/application-development-guide#async-activity-completion)
+
+#### When to use Async Completion
+
+The intended use-case for this feature is when an external system has the final result of a computation, started by an Activity.
+
+Consider sending Heartbeats from the Temporal Client only if the external process is unreliable.
```

Comment:
This is more "Consider _using async activities instead of signals_ only if the external process is unreliable.".

Reply:
"Consider using async activities instead of signals only if the external process is unreliable, and hence might fail to send critical status updates as signals"

I think something like that makes sense.